packages/flutter_tools/test/cache_test.dart

```diff
@@ -77,6 +77,14 @@ void main() {
       verify(artifact2.update());
     });
   });
+
+  testUsingContext('flattenNameSubdirs', () {
+    expect(flattenNameSubdirs(Uri.parse('http://flutter.io/foo/bar')), 'flutter.io/foo/bar');
```

Comment:
this won't work on windows, but otherwise looks fine

Reply:
For this test I am using the MemoryFileSystem (see below), which produces the same paths on Windows and Linux. So, I would expect this to work.

Reply:
Ah, well, that should indeed work then.

What's surprising to me is that given this, it ever worked at all.